### ISSUE TITLE
rom-sql doesn't play nicely with rom-support

### ISSUE THREAD

**Issue by gabrielgiordano** [dry-rb/dry-web-roda#21]
I think rom-support is causing problems when rom-sql is required during boot process.

Steps to reproduce the issue:

``` ruby
# Gemfile
source "https://rubygems.org"
gem "rom",            github: "rom-rb/rom"
gem "rom-sql",        github: "rom-rb/rom-sql"
gem "rom-support",    github: "rom-rb/rom-support"
```

at bash:

`pry`

```
[1] pry(main)> require "bundler/setup"
=> true
[2] pry(main)> require "rom"
=> true
[3] pry(main)> require "rom-sql"
NoMethodError: undefined method `define' for nil:NilClass
from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-support-e76854b0dbd7/lib/rom/support/options.rb:165:in `option'
```

Removing `rom-support` from the `Gemfile` seems to fix the issue.

This is causing problems during boot process after the container tries to boot `component/boot/rom.rb`
when this line is executed

``` ruby
config = ROM::Configuration.new(:sql, container.settings.database_url, extensions: [:error_sql])
```

Here is the stacktrace

```
/home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-support-e76854b0dbd7/lib/rom/support/options.rb:165:in `option': undefined method `define' for nil:NilClass (NoMethodError)
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association/many_to_many.rb:9:in `<class:ManyToMany>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association/many_to_many.rb:4:in `<class:Association>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association/many_to_many.rb:3:in `<module:SQL>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association/many_to_many.rb:2:in `<module:ROM>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association/many_to_many.rb:1:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association.rb:73:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/association.rb:73:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema/associations_dsl.rb:1:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema/associations_dsl.rb:1:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema/dsl.rb:2:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema/dsl.rb:2:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema.rb:35:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/schema.rb:35:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/relation.rb:4:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql/relation.rb:4:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql.rb:11:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-sql-be9248512a7b/lib/rom/sql.rb:11:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/gateway.rb:83:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/gateway.rb:83:in `block in class_from_symbol'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/gateway.rb:81:in `fetch'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/gateway.rb:81:in `class_from_symbol'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/gateway.rb:57:in `setup'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:51:in `block in normalize_gateways'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:45:in `each'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:45:in `each_with_object'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:45:in `normalize_gateways'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:25:in `configure_gateways'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/environment.rb:18:in `initialize'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/configuration.rb:22:in `new'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/bundler/gems/rom-179772b11bd3/lib/rom/configuration.rb:22:in `initialize'
    from /home/gabriel/Workspace/projects/issue/component/boot/rom.rb:8:in `new'
    from /home/gabriel/Workspace/projects/issue/component/boot/rom.rb:8:in `block in <top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-container-0.5.0/lib/dry/container/namespace_dsl.rb:28:in `initialize'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-container-0.5.0/lib/dry/container/mixin.rb:191:in `new'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-container-0.5.0/lib/dry/container/mixin.rb:191:in `namespace'
    from /home/gabriel/Workspace/projects/issue/component/boot/rom.rb:7:in `<top (required)>'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:43:in `require'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:43:in `boot'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:59:in `boot!'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:36:in `block in finalize!'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:35:in `each'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/booter.rb:35:in `finalize!'
    from /home/gabriel/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/dry-system-0.5.1/lib/dry/system/container.rb:249:in `finalize!'
    from /home/gabriel/Workspace/projects/issue/component/boot.rb:6:in `<top (required)>'
    from bin/console:5:in `require_relative'
    from bin/console:5:in `<main>'
```

The exception seems to be coming from here 
https://github.com/rom-rb/rom/blob/master/lib/rom/gateway.rb#L83

After removing the rom-support from the Gemfile ROM seems to boot just fine.

If you guys need, I'm glad to give some help.
Regards

**Comment by timriley** [comment]
Ah, yes, I think this is because we need to make a new release of this gem with our updated Gemfile that we generate in projects. I'll try cut a release today or tomorrow, but in the meantime, could you try generating a project using the master branch of this repo and seeing if you have an issue still?

**Comment by gabrielgiordano** [comment]
The issue was gone when I generated the app using the master branch. Thanks @timriley

**Comment by timriley** [comment]
Great! This is fixed in 0.5.0, releasing in a couple of minutes.
